tests/test_serializers/test_json.py

```diff
@@ -298,8 +303,97 @@ def test_decode_path_error():
             JsonDeserializer.decode_path(value)
 
 
+class TestUnitEncoding:
+    """Tests for custom encoding of Unit types from pint."""
+
+    UNIT_DECODING_TEST_CASES = [
+        (
+            Unit("meter"),
+            CustomEncodedUnitTypes.unit.json_encoding("meter"),
+        ),
+    ]
+
+    UNIT_ENCODING_TEST_CASES = UNIT_DECODING_TEST_CASES
+
+    @staticmethod
+    def test_encode_unit_error():
+        """Test that the JsonSerializer.encode_unit raises an error if no encoding is defined."""
+        # Arrange
+        value = "foo"
+
+        # Act & Assert
+        with pytest.raises(NotImplementedError):
+            JsonSerializer.encode_unit(value)
+
+    @staticmethod
+    @pytest.mark.parametrize("value, expected", UNIT_ENCODING_TEST_CASES)
+    def test_encode_unit(value, expected):
+        """Test that the JsonSerializer.encode_unit encodes values as expected."""
+        # Arrange
+
+        # Act
+        result = JsonSerializer.encode_unit(value)
+
+        # Assert
+        assert result == expected
+
+    @staticmethod
+    @pytest.mark.parametrize("expected, value", UNIT_DECODING_TEST_CASES)
+    def test_decode_unit(value, expected):
+        """Test that the JsonDeserializer.decode_unit decodes Units as expected."""
+        # Arrange
+
+        # Act
+        result = JsonDeserializer.decode_unit(value)
+
+        # Assert
+        assert result == expected
+
+    @staticmethod
+    def test_decode_unit_error():
+        """Test that the JsonDeserializer.decode_unit raises an error if no decoding is defined."""
+        # Arrange
+        value = {"foo": "bar"}
+
+        # Act & Assert
+        with pytest.raises(NotImplementedError):
+            JsonDeserializer.decode_unit(value)
+
+
+class TestUninstantiatedClassEncoding:
+    """Tests for custom encoding of special classes that haven't been instantiated."""
+
+    DECODING_TEST_CASES = [
+        (
+            Unit("meter"),
+            CustomEncodedUnitTypes.unit.json_encoding("meter"),
+        ),
+    ]
+
+    @staticmethod
+    @pytest.mark.parametrize("value, expected", DECODING_TEST_CASES)
+    def test_encode_uninstantiated_class(value, expected):
+        """Test that the JsonSerializer.default encodes special uninstantiated classes as expected."""
+        # Arrange
+
+        # Act
+        result = JsonSerializer.encode_unit(value)
+
+        # Assert
+        assert result == expected
+
+
 def test_round_trip(tmp_path: pathlib.Path):
-    """ Test that a serialized and then deserialized dictionary is unchanged. """
+    """Test that a serialized and then deserialized dictionary is unchanged."""
+    # Define a random class with skip methods
+    class ClassWithSkip:
```

Comment:
Is this used anywhere?

Reply:
nope. dead code. removed

[dc290e8](https://github.com/MORSECorp/snappiershot/pull/96/commits/dc290e8ab8e4e5fb36dd02e60734638decee858c)